pkgs/tools/games/minecraft/optifine/update.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python3 -p python3 python3.pkgs.requests python3.pkgs.lxml
```

Comment:
`-p python3` shouldn't be required since python is already used as the interpreter, and you should make sure to use the current nixpkgs checkout.


This formatting also doesn't change the behaviour of anything but looks nicer IMO:
```suggestion
#!nix-shell -I nixpkgs=./. -p 'python3.withPackages(ps: with ps; [ requests lxml ])'
```

Reply:
`nix-shell` seems to be having a bad time with the spaces... I have thought about this also when writing the script initially but decided it's not worth the trouble.
```
error: syntax error, unexpected invalid token

       at «string»:1:107:

            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ ('python3.withPackages(ps:) (with) (ps;) ([) (requests) (lxml) (])') ]; } ""
             |                                                                                                           ^
```